src/r5py/util/memory_footprint.py

```diff
@@ -72,14 +72,14 @@ def _parse_max_memory_string(max_memory):
     Returns
     -------
     tuple: a tuple containing
-        - value (float): Amount of memory to be allocated in a given unit.
+        - value (int): Amount of memory to be allocated in a given unit.
         - unit (str): The unit of memory.
     """
     try:
         matches = re.match(
             r"^(?P<value>[0-9]+(\.[0-9]+)?)(?P<unit>[^0-9])?$", max_memory
         )
-        value = float(matches["value"])
+        value = int(round(float(matches["value"])))
```

Comment:
round returns an int, casting is not necessary

Reply:
also, it would be nice to be able to supply decimal values: e.g., `1.5G`

Reply:
True. Will convert the logic to what it was